src/Api/ResourceClassResolver.php

```diff
@@ -54,6 +54,9 @@ public function getResourceClass($value, string $resourceClass = null, bool $str
             if (is_subclass_of($type, $resourceClass) && $this->isResourceClass($resourceClass)) {
                 return $type;
             }
+            if ($this->isResourceClass($typeToFind)) {
+                return $typeToFind;
+            }
```

Comment:
May you cache the previous `$this->isResourceClass($typeToFind)` call in a variable instead as this is heavy? Indeed this method can have a big impact on performances.

I'm always kinda scared to modify this file anyway as it can lead to weird behavior. I'm not sure to understand the use case so far, may you give some more explanation? Thanks!

Reply:
I don't see easy way of caching the value without being obliged to call the method each time.
I mean that I can totally enter the and quit the `if` statement without calling this method...
Do you have something to suggest @soyuka ?

Reply:
Hmm, indeed this is why I think this patch is kinda weird (we have `!$this->isResourceClass($typeToFind)` in the condition which means we're in the first part of the condition and that the second needs to be computed afterwards. Maybe this needs a refactor but it is very sensitive ^^.

Reply:
@soyuka I think I found a way to avoid extra call to `isResourceClass($typeToFind)` at least when it is false. WDYT about it ?